File: apps/server/src/api/components/health-check/health-check-controller.ts
import * as httpStatus from 'http-status'
import {
  Controller,
  Example,
  Get,
  Response,
  Route,
  SuccessResponse,
  Tags,
} from 'tsoa'
import { inject, injectable } from 'tsyringe'

import { APIErrorResponse } from '../../shared'
import { HealthCheckMapper } from './health-check-mapper'
import { HealthCheckDTO } from './health-check-models'
import { IHealthCheckService } from './health-check-service'

@injectable()
@Route('health')
@Tags('health-check')
class HealthCheckController extends Controller {
  constructor(
    @inject('HealthCheckService')
    private healthCheckService: IHealthCheckService,
  ) {
    super()
  }

  /**
   * API endpoint used to verify if the service has started up correctly and is ready to accept requests
   */
  @Get('/')
  @SuccessResponse('200', 'OK')
  @Example<HealthCheckDTO>({
    healthy: true,
  })
  @Response<APIErrorResponse>('500', 'Internal Server Error', {
    message: 'Internal Server Error',
    details: { context: undefined, cause: undefined },
    isOperational: false,
  })
  getHealth(): HealthCheckDTO {
    const isHealthy = this.healthCheckService.checkHealth()
    const healthCheckDTO = HealthCheckMapper.toDTO(isHealthy)
    this.setStatus(httpStatus.OK)
    return healthCheckDTO
  }
}

export { HealthCheckController }

File: apps/server2/src/api/components/health/health_check_controller.py
import logging

from dependency_injector.wiring import Provide, inject
from fastapi import APIRouter, Depends
from src.api.components.health.health_check_mapper import HealthCheckMapper
from src.api.components.health.health_check_service import HealthCheckService
from src.services.container_service import Container

logger = logging.getLogger(__name__)


class HealthCheckController(APIRouter):
    def __init__(
        self,
        prefix="/health",
        dependencies=[Depends(Provide[Container.health_check_service])],
    ):
        super().__init__(prefix=prefix, dependencies=dependencies)

        @self.api_route(
            path="",
            methods=["GET"],
            tags=["health-check"],
            description="API endpoint used to verify if the service has started up correctly and is ready to accept requests",  # noqa: E501
        )
        @inject
        async def get_health(
            health_check_service: HealthCheckService = self.dependencies[0],
        ):
            is_healthy = health_check_service.check_health()
            health_check_response = HealthCheckMapper.to_response(is_healthy)
            return health_check_response
